### PR TITLE
Rename search::NumericAttribute::Equal to search::attribute::NumericMatcher.

### PR DESCRIPTION
**Pull request by toregge**
Rename search::NumericAttribute::Range to search::attribute::NumericRangeMatcher.

@geirst : please review
@baldersheim : FYI

### CODE REVIEW

File: searchlib/src/vespa/searchlib/attribute/numeric_matcher.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#pragma once
+
+#include <vespa/searchcommon/common/range.h>
+
+namespace search { class QueryTermSimple; }
+
+namespace search::attribute {
+
+template<typename T>
+class NumericMatcher
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/attribute/numeric_matcher.h:12]: Please add class comment.

File: searchlib/src/vespa/searchlib/attribute/numeric_range_matcher.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+#include <vespa/searchcommon/common/range.h>
+#include <cstddef>
+
+namespace search { class QueryTermSimple; }
+
+namespace search::attribute {
+
+template<typename T>
+class NumericRangeMatcher
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/attribute/numeric_range_matcher.h:13]: Please add class comment.

**Review by geirst — APPROVED** [review]
👍

File: searchlib/src/vespa/searchlib/attribute/numeric_matcher.h (hunk 1 above)
Comment (toregge) [reply searchlib/src/vespa/searchlib/attribute/numeric_matcher.h:16]: Done.

File: searchlib/src/vespa/searchlib/attribute/numeric_range_matcher.h (hunk 2 above)
Comment (toregge) [reply searchlib/src/vespa/searchlib/attribute/numeric_range_matcher.h:17]: Done.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**